src/main/java/org/broadinstitute/hellbender/utils/genotyper/AlleleLikelihoods.java

```diff
@@ -1246,112 +1190,79 @@ public boolean isInformative() {
-        final boolean[] removeIndex = new boolean[sampleReadCount];
-        int removeCount = 0; // captures the number of deletions.
-        int firstDeleted = sampleReadCount;    // captures the first position that was deleted.
-
-        final Iterator<GATKRead> readsToRemoveIterator = readsToRemove.iterator();
-        while (readsToRemoveIterator.hasNext()) {
-            final GATKRead read = readsToRemoveIterator.next();
-            if (indexByRead.containsKey(read)) {
-                final int index = indexByRead.getInt(read);
-                if (firstDeleted > index) {
-                    firstDeleted = index;
-                }
-                removeCount++;
-                removeIndex[index] = true;
-                readsToRemoveIterator.remove();
-                indexByRead.remove(read);
+        for (int a = 0; a < alleleCount; a++) {
+            for (int r = 0; r < evidenceIndicesToKeep.length; r++) {
+                newSampleValues[a][r] = oldSampleValues[a][evidenceIndicesToKeep[r]];
             }
         }
 
-        // Nothing to remove we just finish here.
-        if (removeCount == 0) {
-            return;
-        }
-
-        final int newSampleReadCount = sampleReadCount - removeCount;
-
-        // Now we skim out the removed reads from the read array.
-        final GATKRead[] oldSampleReads = readsBySampleIndex[sampleIndex];
-        final GATKRead[] newSampleReads = new GATKRead[newSampleReadCount];
+        valuesBySampleIndex[sampleIndex] = newSampleValues;
+        evidenceBySampleIndex.set(sampleIndex, newSampleEvidence);
+        evidenceListBySampleIndex[sampleIndex] = null; // reset the unmodifiable list.
+    }
 
-        System.arraycopy(oldSampleReads,0,newSampleReads,0,firstDeleted);
-        Utils.skimArray(oldSampleReads,firstDeleted, newSampleReads, firstDeleted, removeIndex, firstDeleted);
+    /**
+     * Removes those read that the best possible likelihood given any allele is just too low.
+     *
+     * <p>
+     *     This is determined by a maximum error per read-base against the best likelihood possible.
+     * </p>
+     *
+     * @param log10MinTrueLikelihood Function that returns the minimum likelihood that the best allele for a unit of evidence must have
+     * @throws IllegalStateException is not supported for read-likelihood that do not contain alleles.
+     *
+     * @throws IllegalArgumentException if {@code maximumErrorPerBase} is negative.
+     */
+    public void filterPoorlyModeledEvidence(final ToDoubleFunction<EVIDENCE> log10MinTrueLikelihood) {
+        Utils.validateArg(alleles.numberOfAlleles() > 0, "unsupported for read-likelihood collections with no alleles");
 
-        // Update the indices for the extant reads from the first deletion onwards.
-        for (int r = firstDeleted; r < newSampleReadCount; r++) {
-            indexByRead.put(newSampleReads[r], r);
-        }
+        new IndexRange(0, samples.numberOfSamples()).forEach(s -> {
```

Comment:
Stream here are an overkill... it does not make the code simpler:

```
final int numberOfSamples = samples.numberOfSamples();
for (int i = 0; i < numberOfSamples; i++) {
     final List<EVIDENCE> sampleEvidence = evidenceBySampleIndex.get(s);
     final List<EVIDENCE> evidenceToRemove = new ArrayList<>(sampleEvidence.length());
     final int numberOfEvidence = sampleEvidence.size();
     for (int e = 0; e < numberOfEvidence; e++) {
           if (maximuLikelihoodOverAlleles(i, e) < log10MinTrueLikelihood.applyAsDouble(sampleEvidence.get(e)) {
                evidenceToRemove.add(sampleEvidence.get(e);
           }
     }
}
``` 
A few more lines but partly because we cache the table lengths.

Reply:
done